app/authentication/token.py

```diff
@@ -93,19 +88,34 @@ async def is_app_admin(token: str, app: str, error_str: str) -> bool:
         return True
 
 
-async def get_user(token: str = Depends(oauth2_scheme)) -> Tuple[str, str]:
-    """Calls GFW API to authorize user.
-
-    This functions check is user of any level is associated with the GFW
-    app and returns the user ID
-    """
+async def get_user(token: str = Depends(oauth2_scheme)) -> User:
+    """Get the details for authenticated user."""
 
     response: Response = await who_am_i(token)
 
-    if response.status_code == 401 or not (
-        "gfw" in response.json()["extraUserData"]["apps"]
-    ):
+    if response.status_code == 401:
         logger.info("Unauthorized user")
         raise HTTPException(status_code=401, detail="Unauthorized")
     else:
-        return response.json()["id"], response.json()["role"]
+        return User(**response.json())
+
+
+async def get_admin(user: User = Depends(get_user)) -> User:
+    """Get the details for authenticated ADMIN user."""
+
+    if user.role != "ADMIN":
+        raise HTTPException(status_code=401, detail="Unauthorized")
+
+    return user
+
+
+async def get_manager(user: User = Depends(get_user)) -> User:
+    """Get the details for authenticated MANAGER for data-api application or
+    ADMIN user."""
+
+    if user.role != "ADMIN" or not (
+        user.role == "MANAGER" and "data-api" in user.applications
```

Comment:
How does user.applications get populated?  I see that the RW checked-logged returns "extraUserData.apps", not "applications".  https://resource-watch.github.io/doc-api/reference.html#get-lt-base-api-url-gt-auth-check-logged

[Oh, I see this again might be a leftover again from auth/sign-up, which does return an "applications" field, but auth/check-logged doesn't.]

Reply:
I think might be getting confused by that we have two types of "applications" now: https://resource-watch.github.io/doc-api/reference.html#application

I was assuming we'd be using the new one for the "data-api" application, but maybe I got that wrong. @solomon-negusse was that your understanding?

Reply:
OK, but also note that the new 'applications' field is not available from the auth/check-logged call, if that is the one we want to use.

Reply:
ok on ready documentation, this `applications` is really just for RW API keys, we want to use `extraUserData.apps`. I think that's still confusing though because the possible applications in the docs all already use the Data API in some capacity: https://resource-watch.github.io/doc-api/concepts.html#existing-applications

So it feels we're changing the meaning of `application` to also list `data-api` as a separate application, that's used to manage resources for other applications. I think if we want to use the `application` field in the future, we should use it more like RW to control resource management on the Data API. Thoughts @danscales @dmannarino @solomon-negusse @gtempus?